tests/internal/test_serverless.py

```diff
@@ -0,0 +1,40 @@
+import os
+
+import mock
+
+from ddtrace.internal.serverless.mini_agent import maybe_start_serverless_mini_agent
+
+
+@mock.patch("ddtrace.internal.serverless.mini_agent.Popen")
+def test_dont_spawn_mini_agent_if_not_cloud_function(mock_popen):
+    # do not set any GCP environment variables
+
+    maybe_start_serverless_mini_agent()
+
+    mock_popen.assert_not_called()
+
+
+@mock.patch("ddtrace.internal.serverless.mini_agent.Popen")
+def test_spawn_mini_agent_if_deprecated_gcp_function_runtime(mock_popen):
+    os.environ["FUNCTION_NAME"] = "test_function"
```

Comment:
Instead of setting and removing environment variables we can use tests.utils.override_config(). This will better isolate tests. Here's an example: https://github.com/DataDog/dd-trace-py/blob/1.x/tests/appsec/test_appsec_utils.py#L23

Reply:
fixed